internal/build/lifecycle_execution.go

```diff
@@ -550,11 +574,19 @@ func (l *LifecycleExecution) Restore(ctx context.Context, buildCache Cache, kani
 
 func (l *LifecycleExecution) Analyze(ctx context.Context, buildCache, launchCache Cache, phaseFactory PhaseFactory) error {
 	var flags []string
-	args := []string{l.opts.Image.String()}
+	var args []string
+
+	// TODO: a better way to handle the image name (no args are passed to detect)
+	if l.DetectOnly {
+		args = []string{"fakeimage"}
```

Comment:
I could be wrong, but I think in the end the output image name MAY influence the detect process as the selected run image may differ depending on the registry provided and THEN the target data presented to buildpacks may be different.

Reply:
That make sense, resolved